src/sagemaker/model.py

```diff
@@ -375,31 +415,41 @@ def compile(
             job_name,
             framework,
             tags,
+            target_platform_os,
+            target_platform_arch,
+            target_platform_accelerator,
+            compiler_options,
         )
         self.sagemaker_session.compile_model(**config)
         job_status = self.sagemaker_session.wait_for_compilation_job(job_name)
         self.model_data = job_status["ModelArtifacts"]["S3ModelArtifacts"]
-        if target_instance_family.startswith("ml_"):
-            self.image = self._neo_image(
-                self.sagemaker_session.boto_region_name,
-                target_instance_family,
-                framework,
-                framework_version,
-            )
-            self._is_compiled_model = True
-        elif target_instance_family.startswith(INFERENTIA_INSTANCE_PREFIX):
-            self.image = self._inferentia_image(
-                self.sagemaker_session.boto_region_name,
-                target_instance_family,
-                framework,
-                framework_version,
-            )
-            self._is_compiled_model = True
+        if target_instance_family is not None:
+            if target_instance_family.startswith("ml_"):
+                self.image = self._neo_image(
+                    self.sagemaker_session.boto_region_name,
+                    target_instance_family,
+                    framework,
+                    framework_version,
+                )
+                self._is_compiled_model = True
+            elif target_instance_family.startswith(INFERENTIA_INSTANCE_PREFIX):
+                self.image = self._inferentia_image(
+                    self.sagemaker_session.boto_region_name,
+                    target_instance_family,
+                    framework,
+                    framework_version,
+                )
+                self._is_compiled_model = True
+            else:
+                LOGGER.warning(
+                    "The instance type %s is not supported to deploy via SageMaker,"
+                    "please deploy the model manually.",
+                    target_instance_family,
```

Comment:
(I know this isn't your change, but while you're here...)

```suggestion
                    "Deploying instance type %s is not supported via SageMaker."
                    "Please deploy the model manually.",
                    target_instance_family,
```

Reply:
Can we keep old comment? I like it better.

Reply:
changed to `not supported for deployment`